src/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java

```diff
@@ -428,18 +470,43 @@ private void recoverBlockedListeners() {
         }
     }
 
-    private void recoverConnection() throws IOException, InterruptedException {
-        boolean recovering = true;
-        while (recovering) {
+	//Returns true if the connection was recovered, 
+	//false if application initiated shutdown while attempting recovery.  
+    private boolean recoverConnection() throws IOException, InterruptedException {
+        while (!manuallyClosed) //manuallyClosed marked volatile -ensures thread cannot cache value of manuallyClosed.
```

Comment:
The intention was a double check of the manuallyClosed flag.  

It was my assumption that the connection initiation was network io with a timeout.  If that is the case, I don't want to be doing it while holding a lock, as that would cause close/abort calls to unexpectedly block application code.  If that's not a worry, (which maybe it shouldn't be, as a successful close call appears to do network io as well), then I can move it into the synchronized block and remove the double check.